### ISSUE TITLE
[Question] Is it possible to sort project file but ignore `PBXFrameworksBuildPhase` session?

### ISSUE THREAD

**Issue by nelson-dai-17Live** [CocoaPods/Xcodeproj#866]
I'm running the `xcodeproj sort` command line tool, and I'd like to know is it possible to ignore the `PBXFrameworksBuildPhase` session?

I want to keep `PBXFrameworksBuildPhase` as it because the order in this session is important to my project.

Thanks!

**Comment by igor-makarov** [comment]
Are you referring to the list of build phases in the target? The sort explicitly excludes the phase order.

**Comment by nelson-dai-17Live** [comment]
Thank you @igor-makarov for the reply. What I want to achieve is to keep the list in `files = (...);` unchanged.
It is important to the project I'm working on because the project builds failed if the order is changed.

```
/* Begin PBXFrameworksBuildPhase section */
		F5B5C5001774035E00B14D1A /* Frameworks */ = {
			isa = PBXFrameworksBuildPhase;
			buildActionMask = 2147483647;
			files = (
				F54697071AF61A0E00B7985D /* Accelerate.framework in Frameworks */,
				B55A94B517742DEB008F9FBA /* Accounts.framework in Frameworks */,
			);
			runOnlyForDeploymentPostprocessing = 0;
		};
/* End PBXFrameworksBuildPhase section */
```

So I'd like to know if there is / will be any option to keep this content untouched, or can I use `xcodeproj` to build a tool for this goal?
I never use ruby language before, please guide me the direction if I have to build it by myself. Thanks!

**Comment by igor-makarov** [comment]
This is very odd - file order within a frameworks phase shouldn't affect the build. Can you provide more info? What error do you get?

**Comment by nelson-dai-17Live** [comment]
The file order affects `Build Phases -> Link Binary With Libraries` session.
Here are the steps:
1. Close Xcode and run `xcodeproj sort` to sort the project file.
2. Delete `DerivedData` and open Xcode to build project, failed with error (see below).
3. Close Xcode and revert modifications in `PBXFrameworksBuildPhase` sessions.
4. Delete `DerivedData` and open Xcode to build project, success.

It fails to build on simulator, but it works on real device. The error is 
```
Undefined symbols for architecture x86_64:
  "_OPENSSL_ia32cap_P", referenced from:
      _OPENSSL_ia32cap_loc in IJKMediaFramework(cryptlib.o)
      _OPENSSL_cpuid_setup in IJKMediaFramework(cryptlib.o)
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

**Comment by igor-makarov** [comment]
Ok, I've looked at the IJKPlayer podspec and it doesn't look like it was put together by the official developer. It seems that some library isn't listed in the dependency list. Which is why it may be that the error only happens if the order changes.

**Comment by nelson-dai-17Live** [comment]
Yeah, so I think keeping the order may be the easiest way to fix this issue. @igor-makarov do you have any suggestions?

**Comment by igor-makarov** [comment]
You'd need to access the build phase files and reorder them manually after sort. It's a bit complicated so you'd need to get really familiar with the Xcodeproj classes and properties.

**Comment by nelson-dai-17Live** [comment]
Would it be easier to build the sorting function based on `xcodeproj` and bypass `PBXFrameworksBuildPhase` session? If so, is there any sample code or tutorial for newbie like me?

**Comment by igor-makarov** [comment]
Perhaps you may want to override the `#sort` function for the class `PBXFrameworksBuildPhase` in your code and default it to no-op?
